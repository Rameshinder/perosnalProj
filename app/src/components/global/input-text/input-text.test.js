import React from 'react'
import { shallow, mount } from 'enzyme'
import InputText from './index'

describe('InputText', () => {
  const app = shallow(<InputText />)

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })

  it('InputText should render without crashing', () => {
    shallow(<InputText />)
  })

  it('InputText should fire onChange on text change ', () => {
    const spy = jest.spyOn(InputText.prototype, 'handleChange')
    const event = {target: {value: 'spam'}}
    const wrapper = mount(<InputText name='input-text' onChange={jest.fn()} />)
    const textbox = wrapper.find('[name="input-text"]').at(1)
    textbox.simulate('change', event)
    expect(spy).toHaveBeenCalled()
  })

  it('InputText should fire onclick on textbox click ', () => {
    const spy = jest.spyOn(InputText.prototype, 'handleClick')
    const event = {target: {value: 'spam'}}
    const wrapper = mount(<InputText name='input-text' onClick={jest.fn()} />)
    const textbox = wrapper.find('[name="input-text"]').at(1)
    textbox.simulate('click', event)
    expect(spy).toHaveBeenCalled()
  })

  it('onkeypress event for InputText', () => {
    const spy = jest.spyOn(InputText.prototype, 'handleKeypress')
    const event = {target: {value: 'spam'}}
    const wrapper = mount(<InputText name='input-text' onKeyPress={jest.fn()} />)
    const textbox = wrapper.find('[name="input-text"]').at(1)
    textbox.simulate('keypress', event)
    expect(spy).toHaveBeenCalled()
  })

  it('onkeyup event for InputText', () => {
    const spy = jest.spyOn(InputText.prototype, 'handleKeyup')
    const event = {target: {value: 'spam'}}
    const wrapper = mount(<InputText name='input-text' onKeyUp={jest.fn()} onkeyup={jest.fn()} />)
    const textbox = wrapper.find('[name="input-text"]').at(1)
    textbox.simulate('keyup', event)
    expect(spy).toHaveBeenCalled()
  })
})
