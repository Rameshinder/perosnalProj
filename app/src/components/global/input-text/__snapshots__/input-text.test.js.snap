// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputText
    autocapsformat={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <input
          aria-label={undefined}
          aria-required={undefined}
          data-required={undefined}
          data-test-id={undefined}
          id={undefined}
          maxLength={undefined}
          name={undefined}
          onChange={[Function]}
          onClick={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          placeholder={undefined}
          readOnly={undefined}
          tabIndex={0}
          type={undefined}
          value=""
        />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": undefined,
          "aria-required": undefined,
          "data-required": undefined,
          "data-test-id": undefined,
          "id": undefined,
          "maxLength": undefined,
          "name": undefined,
          "onChange": [Function],
          "onClick": [Function],
          "onKeyPress": [Function],
          "onKeyUp": [Function],
          "placeholder": undefined,
          "readOnly": undefined,
          "tabIndex": 0,
          "type": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <input
            aria-label={undefined}
            aria-required={undefined}
            data-required={undefined}
            data-test-id={undefined}
            id={undefined}
            maxLength={undefined}
            name={undefined}
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder={undefined}
            readOnly={undefined}
            tabIndex={0}
            type={undefined}
            value=""
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": undefined,
            "aria-required": undefined,
            "data-required": undefined,
            "data-test-id": undefined,
            "id": undefined,
            "maxLength": undefined,
            "name": undefined,
            "onChange": [Function],
            "onClick": [Function],
            "onKeyPress": [Function],
            "onKeyUp": [Function],
            "placeholder": undefined,
            "readOnly": undefined,
            "tabIndex": 0,
            "type": undefined,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
